src/panels/lovelace/cards/hui-sensor-card.ts

```diff
@@ -0,0 +1,375 @@
+import { HomeAssistant } from "../../../types";
+import { fireEvent } from "../../../common/dom/fire_event";
+
+import computeStateName from "../../../common/entity/compute_state_name";
+import stateIcon from "../../../common/entity/state_icon";
+
+import "../../../components/ha-card";
+import "../../../components/ha-icon";
+
+interface Config extends LovelaceCardConfig {
+  entity: string;
+  name?: string;
+  icon?: string;
+  graph?: string;
+  unit?: string;
+  detail?: number;
+  hours_to_show?: number;
+}
+
+class HuiSensorCard extends LitElement implements LovelaceCard {
+  public hass?: HomeAssistant;
+  private _config?: Config;
+
+  static get properties(): PropertyDeclarations {
+    return {
+      hass: {},
+      _config: {},
+    };
+  }
+
+  public setConfig(config: Config): void {
+    if (!config.entity || config.entity.split(".")[0] !== "sensor") {
+      throw new Error("Specify an entity from within the sensor domain.");
+    }
+
+    const cardConfig = {
+      detail: 1,
+      hours_to_show: 24,
+      ...config,
+    };
+
+    cardConfig.hours_to_show = Number(cardConfig.hours_to_show);
+    cardConfig.detail =
+      cardConfig.detail === 1 || cardConfig.detail === 2
+        ? cardConfig.detail
+        : 1;
+
+    this._config = cardConfig;
+  }
+
+  public getCardSize(): number {
+    return 3;
+  }
+
+  protected render(): TemplateResult {
+    if (!this._config || !this.hass) {
+      return html``;
+    }
+
+    const stateObj = this.hass.states[this._config.entity];
+    return html`
+      ${this.renderStyle()}
+      <ha-card @click="${this._handleClick}">
+        ${
+          !stateObj
+            ? html`
+                <div class="not-found">
+                  Entity not available: ${this._config.entity}
+                </div>
+              `
+            : html`
+                <div class="flex">
+                  <div class="icon">
+                    <ha-icon
+                      .icon="${this._config.icon || stateIcon(stateObj)}"
+                    ></ha-icon>
+                  </div>
+                  <div class="header">
+                    <span class="name"
+                      >${this._config.name || computeStateName(stateObj)}</span
+                    >
+                  </div>
+                </div>
+                <div class="flex info">
+                  <span id="value">${stateObj.state}</span>
+                  <span id="measurement"
+                    >${
+                      this._config.unit ||
+                        stateObj.attributes.unit_of_measurement
+                    }</span
+                  >
+                </div>
+                <div class="graph">
+                  <div>
+                    ${
+                      this._config.graph !== "none" &&
+                      stateObj.attributes.unit_of_measurement
+                        ? until(
+                            this._getHistory().then((history) => {
```

Comment:
So eh, you're not caching this… So we're going to fetch the history on each change of `hass`, even if it wasn't even this entity that changed?